api/src/main/java/au/org/aodn/nrmn/restapi/validation/process/RawValidation.java

```diff
@@ -40,169 +40,157 @@
+
+                val diver = (Diver) values.get("Diver").orElseGet(null);
+                val longitude = (Double) values.get("Longitude").orElseGet(null);
+                val latitude = (Double) values.get("Latitude").orElseGet(null);
+
+                val splitDepth = values.get("Depth").orElseGet(null).toString().split("\\.");
+                val depth = Integer.parseInt(splitDepth[0]);
+
+                Optional<Integer> survey_num = splitDepth.length == 1 ? Optional.empty()
+                                : Optional.of(Integer.parseInt(splitDepth[1]));
+
+                val method = (Integer) values.get("Method").orElseGet(null);
+                val block = (Integer) values.get("Block").orElseGet(null);
+                val species = Optional.ofNullable((ObservableItem) values.get("Species").orElse(null));
+
+                Optional<UiSpeciesAttributes> mayBeSpeciesAttributes = species.isPresent()
+                                ? obsRepo.getSpeciesAttributesById(species.get().getObservableItemId())
+                                : Optional.empty();
+
+                val code = (String) values.get("Code").orElseGet(null);
+
+                val vis = (Optional<Integer>) values.get("Vis").orElse(Optional.empty());
+                val total = (Integer) values.get("Total").orElseGet(null);
+                val inverts = (Optional<Integer>) values.get("Inverts").orElse(Optional.empty());
+                val direction = (Directions) values.get("Direction").orElseGet(null);
+                val measureJson = (java.util.Map<Integer, Integer>) values.get("MeasureJson").orElseGet(null);
+
+                val ref = (StagedRow) values.get("Ref").orElseGet(null);
+
+                val rowFormatted = new StagedRowFormatted();
+                rowFormatted.setId(ref.getId());
+                rowFormatted.setDate(date);
+                rowFormatted.setTime(time);
+                rowFormatted.setSite(site);
+                rowFormatted.setDiver(diver);
+                rowFormatted.setLongitude(longitude);
+                rowFormatted.setLatitude(latitude);
+                rowFormatted.setDepth(depth);
+                rowFormatted.setSurveyNum(survey_num);
+                rowFormatted.setMethod(method);
+                rowFormatted.setBlock(block);
+                rowFormatted.setSpecies(species);
+                rowFormatted.setVis(vis);
+                rowFormatted.setCode(code);
+                rowFormatted.setDirection(direction);
+                rowFormatted.setTotal(total);
+                rowFormatted.setInverts(inverts.orElse(0));
+                rowFormatted.setMeasureJson(measureJson);
+                rowFormatted.setRef(ref);
+                rowFormatted.setSpeciesAttributesOpt(mayBeSpeciesAttributes);
+
+                if (isExtendedSizing) {
+                        val isInvertSizing = (Optional<Boolean>) values.get("IsInvertSizing").orElse(Optional.empty());
```

Comment:
??  I thought we weren't using optionals any more  - doesn't this throw a class cast exception?

Reply:
And this value is actually optional when isextendedsizing is false

Reply:
Changed the validation step but not the subsequent uses. Converting back to draft.